FORMS.md

```diff
@@ -0,0 +1,243 @@
+- `name`: Name of the Input. Submitted with the form as part of a name/value pair.
+- `readonly`: Value is not editable.
+- `required`: A value is required.
+- `type`: Type of Input.
+- `value`: Initial value of the Input.
+
+There are specific attributes that will need to be passed to individual Input types as well. <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input">This chart from MDN</a> provides an excellent breakdown.
+
+### States
+
+Many of these states are assigned or applicable to the `<input>` element. However, those Input states may be better reflected at the Field level. For example: dimming Field.Label for `disabled` elements; showing Field.Error only when an Input is `invalid`.
+
+- `active`: When the Input is activated by the user.
+- `autofill`: When an Input has been auto-filled by the browser.
+- `blank`: When an Input is empty.
+- `checked`: When Inputs like Radio and Checkbox are checked.
+- `dir`: The reading direction currently applied.
+- `disabled`: When an Input is disabled.
+- `enabled`: When an Input is enabled (rather than disabled). This is the default state.
+- `focus-within` When a child has `focus`.
+- `focus`: When the Input has focus.
+- `hover`: When the Input is hovered by a pointing device.
+- `in-range` When an Input is within the range constrictions. e.g. a slider.
+- `indeterminate`: When Inputs are in an indeterminate state. e.g. A "parent" checkbox for a group of mixed checked/unchecked Inputs.
+- `invalid`: When an Input has invalid contents/values.
+- `optional`: When an Input is optional.
+- `out-of-range`: When an Input is outside of range constrictions.
+- `placeholder-shown`: When placeholder text is visible.
+- `read-only`: When an Input cannot be changed by the user.
+- `read-write`: When an Input is user-editable (rather than read-only). This is the default state.
+- `required`: When an Input is required.
+- `target-within`: When a child is a `target`.
+- `target`: When the component is the target of a URL. e.g. `site.com/form#firstname`
+- `user-invalid`: When an Input is invalid, but only after a user has interacted with it.
+- `valid`: When an Input has valid contents/values.
+
+### Content
+
+- FieldLabel: a text label for the field (element may be `<label>` or `<legend>` depending on Field input)
+  - The Label also indicates the Field's optional/required status
+- FieldHint: context for the field
+- Input: the UI control
+- FieldValidation: any error or success state messaging for the field
+
+### Psuedo-structure
+
+```html
+// via props
+
+<Field
+  label="A field"
+  hint="This is a field you should fill in."
+  type="text"
+  name="a-field"
+  optional
+  disabled
+  autocomplete="fname"
+/>
+
+
+// via composition
+
+<Field>
+  <Field.Label>A field</Field.Label>
+  <Field.Hint>This is a field you should fill in.</Field.Hint>
+  <Input/>
+  <Field.Validation type="error">This is an error message.</Field.Validation>
+</Field>
```

Comment:
I know this is a pseudo structure, so let me know if this feedback isn't useful right now.

I think this is an interesting use of static's, on one hand, I like that it's very "HTML-like" in how you use it, on the other though, I fear it could increase developer cognitive load. I think props might be a better choice here, for example:

```html
<Field 
  label="A field" 
  hint="This is a field you should fill in." 
  hasError={true} 
  error="This is an error message."
>
  <Input/>
</Field>
```

Reply:
This is my preferred interface for `Field` as well - seems friendlier than composition in this case.

Reply:
Good feedback - we are going to be balancing the flexibility of composition based APIs and the simplicity of primitive prop based APIs. Keep in mind that props can also be render functions or trees of components so composition can still occur outside the children API as part of non primitive props.

In the example shared above using composition for the label would allow us to provide a screen reader only label rather than splitting that instruction across two primitive props `label` and `screenReaderOnlyLabel`.

Reply:
Part of the goal is that developers/designers should not be specifying most of this behavior or content. Ex, `label` and `screenReaderOnlyLabel` should not be associated or populated manually. The content of one should dictate the content of the other, and the behavior for both should be standardized.